private/bufpkg/bufplugin/bufplugindocker/docker_test.go

```diff
@@ -193,22 +123,38 @@ func createClient(t testing.TB, options ...ClientOption) Client {
 	return dockerClient
 }
 
-func buildDockerPlugin(t testing.TB, dockerClient Client, dockerfilePath string, pluginIdentity string, pluginVersion string) (*BuildResponse, error) {
+func buildDockerPlugin(t testing.TB, dockerfilePath string, pluginIdentity string) (string, error) {
```

Comment:
Removed code which used to build docker images in favor of just shelling out to the docker command to run some basic tests. This still allows us to keep coverage up for issues seen when pushing images, etc.

Reply:
At this point we're effectively testing the docker cli, do we still find these tests useful?

https://github.com/bufbuild/buf/blob/34fdcfa897728de29e527ba4e30c06d856ab654c/private/bufpkg/bufplugin/bufplugindocker/docker.go#L205-L208

Did the previous tests require you to have docker cli or the docker engine installed?

Reply:
Yeah they did. We still exercise tests where we simulate errors from the docker registry push, etc. I think it is still useful to keep around.